### DISCUSSION TITLE
Query String and Browser Back when  $queryString property present

### DISCUSSION THREAD

**Discussion by vakisp** [livewire/livewire#4470] [7. Bugs (archived)]
Hi,

After several, very long, days trying to figure out what was happening I think there may be a bug with the query string implementation. It may be related to the following but can't really be sure:

QueryString on v2.6 #3734 
Query filters not persisted when navigating back #3767 

I think it may also be related to changes made to handle the back-forward browser cache implementation:

[Fix] back button cache issues #2747

I initially thought it was an issue with the WithPagination trait but after some trial and error and debugging realised it was just the $querystring property causing the issue.

If there there is a $querystring property either in a component or indirectly via a trait such as WithPagination then the browser back button behaviour breaks in Safari, chrome seems fine. 

What happens is when back is pressed it seems to re-render the component correctly and then bounces back to the original page and re-renders that page again (that could be just the DOM diffing doing that as there does not appear to be any extra network requests).

My page has a list of items at the bottom, when you click one it redirects to the same URL with the item id in the url as a querystring parameter which causes the page to render the item details. This works fine until you press the back button, on safari, instead of going to the previous item it bounces you back to the current one.

I've managed to create a simple workaround for my situation as follows:

1 - Modified the WithPagination trait to always return an empty array as follows:

    public function queryStringWithPagination()
    {
        foreach ($this->paginators as $key => $value) {
            $this->$key = $value;
        }

        return [];
    }

2 - To ensure I have the right page in the header when redirecting I add it to the redirect as follows:

.....

$params = [ .... ]

if ($this->page > 1) {
        $params['page'] = $this->page;
}
return redirect()->to(route('myRoute', $params));

.....

What confused the situation further is that in production, using Apache, my workaround works, however in development using php artisan serve the following is required in the page header (only for safari), I assume caching is not handled in the same way in both servers:

<head>
      <script>
            // Required to manage safari browser back button issues
            (function () {
                  window.onpageshow = function(event) {
                        if (event.persisted) {
                                    window.location.reload();

                        }
                  };
            })();
      </script>

....

So not sure what is going on here but I will try and resolve it and if I do I will update you all, in the mean time I just wanted to raise this here incase others are having the same issue, or if someone can help me if its just me doing something wrong.

Kind regards,

Vakis

**Comment by joshhanley** [comment]
@vakisp thanks for reporting! Any chance you could create a small example repo which reproduces the issue with as little code as possible so we can investigate it?

Thanks!
